migration-engine/connectors/sql-migration-connector/src/sql_schema_calculator.rs

```diff
@@ -89,219 +93,149 @@ impl<'a> SqlSchemaCalculator<'a> {
         }
     }
 
-    fn calculate_model_tables(&self) -> SqlResult<Vec<ModelTable>> {
-        datamodel_helpers::walk_models(self.data_model)
-            .map(|model| {
-                let columns = model
-                    .fields()
-                    .flat_map(|f| match f.field_type() {
-                        TypeRef::Base(_) => Some(sql::Column {
+    fn calculate_model_tables<'iter>(
+        &'iter self,
+    ) -> impl Iterator<Item = SqlResult<(ModelRef<'a>, sql::Table)>> + 'iter {
+        datamodel_helpers::walk_models(self.data_model).map(move |model| {
+            let columns = model
+                .fields()
+                .flat_map(|f| match f.field_type() {
+                    TypeRef::Base(_) => Some(sql::Column {
+                        name: f.db_name().to_owned(),
+                        tpe: column_type(&f),
+                        default: migration_value_new(&f),
+                        auto_increment: {
+                            match f.default_value() {
+                                Some(DefaultValue::Expression(ValueGenerator {
+                                    name: _,
```

Comment:
Could you use .. here?

Reply:
yes most definitely, we can even simplify this whole match to a `matches!` call. I'll do that.

Reply:
Simplified to `matches!(f.default_value(), Some(DefaultValue::Expression(ValueGenerator { generator: ValueGeneratorFn::Autoincrement, .. })))`